x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/kerberos/KerberosRealmTests.java

```diff
@@ -125,14 +128,27 @@ public void testKerberosRealmWithInvalidKeytabPathConfigurations() throws IOExce
             expectedErrorMessage = "configured service key tab file [" + keytabPath + "] does not exist";
             break;
         case "keytabPathWithNoReadPermissions":
-            filePerms = PosixFilePermissions.fromString("---------");
-            final String keytabFileName = randomAlphaOfLength(5) + ".keytab";
-            final FileAttribute<Set<PosixFilePermission>> fileAttributes = PosixFilePermissions.asFileAttribute(filePerms);
-            try (SeekableByteChannel byteChannel = Files.newByteChannel(dir.resolve(keytabFileName),
-                    EnumSet.of(StandardOpenOption.CREATE_NEW, StandardOpenOption.WRITE), fileAttributes)) {
-                byteChannel.write(ByteBuffer.wrap(randomByteArrayOfLength(10)));
+            final String fileName = randomAlphaOfLength(5);
+            final Path keytabFilePath = Files.createTempFile(dir, fileName, ".keytab");
+            Files.write(keytabFilePath, randomAlphaOfLength(5).getBytes(StandardCharsets.UTF_8));
+            final Set<String> supportedAttributes = keytabFilePath.getFileSystem().supportedFileAttributeViews();
+            if (supportedAttributes.contains("posix")) {
+                final PosixFileAttributeView fileAttributeView = Files.getFileAttributeView(keytabFilePath, PosixFileAttributeView.class);
+                fileAttributeView.setPermissions(PosixFilePermissions.fromString("---------"));
+            } else if (supportedAttributes.contains("acl")) {
+                final UserPrincipal principal = Files.getOwner(keytabFilePath);
+                final AclFileAttributeView view = Files.getFileAttributeView(keytabFilePath, AclFileAttributeView.class);
+                final AclEntry entry = AclEntry.newBuilder()
+                        .setType(AclEntryType.DENY)
+                        .setPrincipal(principal)
+                        .setPermissions(AclEntryPermission.READ_DATA, AclEntryPermission.READ_ATTRIBUTES).build();
+                final List<AclEntry> acl = view.getAcl();
+                acl.add(0, entry);
+                view.setAcl(acl);
+            } else {
+                throw new UnsupportedOperationException("Unsupported file attributes for this test");
```

Comment:
I think this message should be a bit more clear.
Can you include: 
- the path
- the supportedAttributes
- some explanation about what attributes we're looking for

It can just be `"Don't know how to make file {} non-readable on a filesystem with attributes {}"`